File: pset1 - Intro to C/cash problem/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int get_pos_float(string prompt);

int main(void)
{
    int qtr = 25;
    int dm = 10;
    int nic = 5;
    int pen = 1;
    int c = get_pos_float("Change Owed: ");
    int count = 0;
    while (c >= qtr)
    {
        count++;
        c -= qtr;
    }
    while(c >= dm)
    {
        count++;
        c-=dm;
    }
    while(c >= nic)
    {
        count++;
        c-=nic;
    }
    while(c >= pen)
    {
        count++;
        c-=pen;
    }
    printf("%i\n", count);
}

int get_pos_float(string prompt)
{
    float n;
    do
    {
         n = round(get_float("%s", prompt) * 100);
    }
    while (n < 0);
    return n;
}

File: pset6 - Intro to Python/cash/cash.py
from cs50 import get_float
from cs50 import get_int
from cs50 import get_string


# defining our function to get positive inputs only
def get_pos_float(prompt):
    while True:
        n = get_float(prompt) * 100
        if n > 0:
            break
    return n


# defining the coins as variables
qtr = 25
dm = 10
nic = 5
pen = 1

# get float and convert to int
c = get_pos_float("Change Owed: ")
cash = int(c)

# using modulo instead of while loops to calculate how many coins for change as per the challenge in the problem set
q = (cash // qtr)
d = ((cash % qtr) // dm)
n = ((cash % qtr) % dm) // nic
p = ((cash % qtr) % dm) % nic

# print total number of least coins
x = q + d + n + p
print(x)